src/components/docs-anchor/index.tsx

```diff
@@ -48,15 +42,15 @@ const DocsAnchor = ({
   if (adjustedHref !== href) {
     return (
       <Link href={adjustedHref}>
```

Comment:
📌  Random related thing: IIRC `next/link` _can_ be used for external links (it seems to work I think?) but doesn't seem to be recommended (wording [in next/link docs](https://next-js.ir/docs/api-reference/next/link) is:

> External URLs, and any links that don't require a route navigation using /pages, don't need to be handled with Link; use the anchor tag for such cases instead.

I wonder whether the "don't need" wording there has significant performance implications... or whether we could drop `MaybeInternalLink` altogether, and use `next/link` all the time to avoid messy logic? 🤔

Reply:
I think @dstaley can speak more to this because he did some related work on one of our properties (maybe dotcom?), but AFAIK it's OK for us to always use `next/link`! And we actually have [a ticket in the backlog to remove `MaybeInternalLink`](https://app.asana.com/0/1202097197789424/1202129912162168/f) when there's time. 😊

Reply:
Oh other thought here - do we need `passHref` here on `next/link`?

ref: <https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-custom-component-that-wraps-an-a-tag>

Reply:
Hmm that `code` example is a good point. I wonder how many of these manual `<a>` elements need to be replaced with `InlineLink`/`MDXInlineLink`, and if `InlineLink` needs to be updated to replace the `text` prop with `children`? Would mean not using `Text` inside of `InlineLink`, but I think we could probably use CSS composition in place of that. Thoughts?

Reply:
@zchsh The goal with this PR is just to address style inconsistencies with tutorials anchor links, so I'm hesitant to make more changes to this docs-anchor without having full context to how this original docs anchor component was set up. Should we make a follow up task?

Reply:
@kendallstrautman For the style issues, absolutely! We've got 🎟️ [[code-link] Visited style refinement](https://app.asana.com/0/1202097197789424/1202331881539237/f) which I think covers it.

For `passHref`, I think it should be added in this PR, as otherwise it's a regression (previously `href`s were rendered into the DOM without the need for `passHref` since we had plain `<a />` element rather than "custom" component).

I think all we need is:

```suggestion
      <Link href={adjustedHref} passHref>
```

Reply:
@zchsh do our suggested changes conflict? Here's what I'm suggesting in another thread: https://github.com/hashicorp/dev-portal/pull/469#discussion_r882979173

Reply:
So y'all think I should keep the `a` with the custom style defs? I'm just so confused now. Maybe I should just duplicate the styles for tutorials and not touch this docs component.

Reply:
@zchsh @ashleemboyer I reverted updates to the docs anchor component and added a TODO to refactor and use this new mdx link component.

Reply:
☝️ I'm good with that - re-affirming my approval on this PR ✅ apologies for the confusion!